crates/bitcoin-da/src/monitoring.rs

```diff
@@ -0,0 +1,502 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use tokio::sync::{Mutex, RwLock};
+use tokio::time::interval;
+use tracing::{error, info, instrument};
+
+use bitcoin::{BlockHash, Transaction, Txid};
+use bitcoincore_rpc::json::GetTransactionResult;
+use bitcoincore_rpc::{Client, RpcApi};
+use thiserror::Error;
+
+use crate::service::FINALITY_DEPTH;
+
+// Todo pass down lower value for test
+// const DEFAULT_CHECK_INTERVAL: Duration = Duration::from_secs(60);
+const DEFAULT_CHECK_INTERVAL: Duration = Duration::from_millis(10);
+const DEFAULT_HISTORY_LIMIT: usize = 1_000; // Keep track of last 1k txs
```

Comment:
putting size limitation on the history is probably helpful as well.

Reply:
yeah the size of all txs in history, might be unnecessary though.

but for the prover for example

assume at high traffic for a long time 

400kb for 2 tx groups (commit + reveal)

totals at 195 MB

hmm, that's fine actually

or is the history just txid?